packages/editor/src/editor/browser/cms/useTextSelection.test.ts

```diff
@@ -0,0 +1,104 @@
+import { describe, test, expect } from 'vitest';
+import { renderHook } from '@testing-library/react';
+import useTextSelection, { type InputTextAreaRef } from './useTextSelection';
+import { act } from 'react';
+
+const mockRef = (value: string, selectionStart: number | null, selectionEnd: number | null) => {
+  return {
+    current: {
+      value,
+      selectionStart,
+      selectionEnd
+    } as HTMLInputElement | HTMLTextAreaElement
+  } as InputTextAreaRef;
+};
+
+describe('useTextSelection', () => {
```

Comment:
I would use this `describe`'s in another way.
You now use it a global group with every test in it. But the test file (which is also part of the test name) has the useTextSelection also in its name. So it is duplicated now.

However in this case I think grouping tests can make sense, as the hook gives you multiple functions.
So I would group the tests after them `selection`, `getSelectedText`, ...